lib/src/sql/value/value.rs

```diff
@@ -998,100 +1011,615 @@ impl Value {
+				Ok(v) => Ok(v),
+				// Ths string is not a datetime
+				Err(_) => Err(Error::ConvertTo {
+					from: self,
+					into: "datetime".into(),
+				}),
 			},
-			Kind::Record(t) => match self.is_type_record(t) {
-				true => self,
-				_ => Value::None,
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "datetime".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to a `Duration`
+	pub fn convert_to_duration(self) -> Result<Duration, Error> {
+		match self {
+			// Durations are allowed
+			Value::Duration(v) => Ok(v),
+			// Attempt to parse a string
+			Value::Strand(ref v) => match Duration::try_from(v.as_str()) {
+				// The string can be represented as a duration
+				Ok(v) => Ok(v),
+				// Ths string is not a duration
+				Err(_) => Err(Error::ConvertTo {
+					from: self,
+					into: "duration".into(),
+				}),
 			},
-			Kind::Geometry(t) => match self.is_type_geometry(t) {
-				true => self,
-				_ => Value::None,
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "duration".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to a `Bytes`
+	pub fn convert_to_bytes(self) -> Result<Bytes, Error> {
+		match self {
+			// Bytes are allowed
+			Value::Bytes(v) => Ok(v),
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "bytes".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to an `Object`
+	pub fn convert_to_object(self) -> Result<Object, Error> {
+		match self {
+			// Objects are allowed
+			Value::Object(v) => Ok(v),
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "object".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to an `Array`
+	pub fn convert_to_array(self) -> Result<Array, Error> {
+		match self {
+			// Arrays are allowed
+			Value::Array(v) => Ok(v),
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "array".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to an `Geometry` point
+	pub fn convert_to_point(self) -> Result<Geometry, Error> {
+		match self {
+			// Geometry points are allowed
+			Value::Geometry(Geometry::Point(v)) => Ok(v.into()),
+			// An array of two floats are allowed
+			Value::Array(ref v) if v.len() == 2 => match v.as_slice() {
+				// The array can be represented as a point
+				[Value::Number(v), Value::Number(w)] => Ok((v.to_float(), w.to_float()).into()),
+				// The array is not a geometry point
+				_ => Err(Error::ConvertTo {
+					from: self,
+					into: "point".into(),
+				}),
 			},
+			// Anything else raises an error
+			_ => Err(Error::ConvertTo {
+				from: self,
+				into: "point".into(),
+			}),
+		}
+	}
+
+	/// Try to convert this value to a Record or `Thing`
+	pub fn convert_to_record(self) -> Result<Thing, Error> {
+		match self {
+			// Records are allowed
+			Value::Thing(v) if self.is_record() => Ok(v),
```

Comment:
Isn't the `if self.is_record()` redundant here?

(for reference, `is_record` was previously defined as the following. I'll check again just in case it was redefined)
```rust
	/// Check if this Value is a Thing
	pub fn is_record(&self) -> bool {
		matches!(self, Value::Thing(_))
	}
```